src/main/java/underdogs/devbie/chat/domain/ChatNameFactory.java

```diff
@@ -21,27 +24,42 @@
     }
 
     public static ChatName createNonOverlappingName(ChatNames existNameInChatRoom) {
+        Set<ChatName> chatNames = removeExistNameInChatRoom(existNameInChatRoom);
+
+        ChatName nonOverlappingName = extractRandom(chatNames);
+
+        nonOverlappingName.setColor(extractRandom(colors));
+
+        return nonOverlappingName;
+    }
+
+    private static Set<ChatName> removeExistNameInChatRoom(ChatNames existNameInChatRoom) {
         Set<ChatName> chatNames = new HashSet<>();
         for (ChatName chatName : names) {
             chatNames.add(ChatName.of(chatName.getChatName(), null));
         }
 
         chatNames.removeAll(existNameInChatRoom.getChatNames());
-
-        ChatName nonOverlappingName = chatNames.iterator().next();
-
-        TitleColor nonOverlappingColor = createNonOverlappingColor(existNameInChatRoom.getChatNames());
-
-        nonOverlappingName.setColor(nonOverlappingColor);
-
-        return nonOverlappingName;
+        return chatNames;
     }
 
-    private static TitleColor createNonOverlappingColor(Set<ChatName> existNameInChatRoom) {
-        Set<TitleColor> existColorInChatRoom = existNameInChatRoom.stream()
-            .map(ChatName::getColor)
-            .collect(Collectors.toSet());
+    private static <T> T extractRandom(Set<T> set) {
+        int size = set.size();
+        if (size == 0) {
+            throw new IndexOutOfBoundsException("컬렉션의 크기가 0입니다.");
+        }
+        int randomIndex = new Random().nextInt(size);
+        int i = 0;
+        for (T item : set) {
+            if (i == randomIndex) {
+                return item;
+            }
+            i++;
+        }
+        throw new IllegalArgumentException("도출될 수 없는 결과입니다.");
+    }
```

Comment:
이런식으로 stream을 활용해보면 더 좋을 것 같아요
```suggestion
   private static <T> T extractRandom(Set<T> set) {
       int randomIndex = new Random().nextInt(set.size());
       
       return set.stream()
               .skip(randomIndex)
               .findFirst()
               .orElseThrow();
   }
```

Reply:
skip 메서드가 유용하네요!

어제 Stream filter를 사용하려다가 filter는 처음부터 끝까지 다 검사할텐데 하고 for each가 괜찮겠다 라고 생각했거든요.

꿀팁감사해요